Automattic/wp-calypso
# Signup: Reduxify the ProgressStore with backward compatible API

Work in progress.

Step 2 of #6709.

This change converts SignupProgressStore to be powered by Redux instead of Flux, much like how the SignupDependencyStore was reduxified in #6124 (and #25510). 

The ProgressStore, among other things, is responsible for tracking and persisting the user's progression through our various signup flows.

All existing signup flow behavior should remain identical.

# Test Instructions
1. Audit the code.
2. Spin up this branch locally.
3. While logged out, go through user registration flow from `/`.
4. At any point in the registration flow, try refreshing the page. Verify that the interface restores your progress.
5. At any point in the registration flow, try closing and reopening your window. Verify that the interface restores your progress.
4. (For e2e testers) Try going through the registration flow for Jetpack and WooCommerce.
